src/helpers/coordinates.ts

```diff
@@ -53,37 +93,21 @@ function isCharADigit(char: string) {
  * Note: it also accepts lowercase coordinates, but not fixed references
  */
 export function toCartesian(xc: string): Position {
-  xc = xc.trim();
-
-  let letterPart = "";
-  let numberPart = "";
-  let i = 0;
+  const chars = new TokenizingChars(xc);
 
-  // Process letter part
-  if (xc[i] === "$") i++;
-  while (i < xc.length && isCharALetter(xc[i])) {
-    letterPart += xc[i++];
-  }
+  consumeSpaces(chars);
+  const letterPart = consumeLetters(chars);
 
-  if (letterPart.length === 0 || letterPart.length > 3) {
-    // limit to max 3 letters for performance reasons
+  if (letterPart === -1 || !chars.current) {
     throw new Error(`Invalid cell description: ${xc}`);
   }
 
-  // Process number part
-  if (xc[i] === "$") i++;
-  while (i < xc.length && isCharADigit(xc[i])) {
-    numberPart += xc[i++];
-  }
-
-  if (i !== xc.length || numberPart.length === 0 || numberPart.length > 7) {
-    // limit to max 7 numbers for performance reasons
-    throw new Error(`Invalid cell description: ${xc}`);
-  }
+  const num = consumeDigits(chars);
```

Comment:
case where consumeDigits returns -1 is not handled

Reply:
well, it turns out `toCartesian("A0")` was "valid" before this commit. At least it's valid from the regexp point of view. And there's bloody test failing if I handle -1 and rejects. I didn't bother changing this behavior, especially in stable